### PR TITLE
[DOCS] Stop linking to indices.templates.html page

### PR DESCRIPTION
**Pull request by pquentin**
Right now, the Elasticsearch Python client docs about index templates (legacy and composable) link to https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates.html. This page used to look like this: https://www.elastic.co/guide/en/elasticsearch/reference/7.7/indices-templates.html but is no longer useful.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.exists_index_template.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "indices.exists_index_template":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-templates.html",
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/index-templates.html",
```
Comment (pquentin) [inline rest-api-spec/src/main/resources/rest-api-spec/api/indices.exists_index_template.json:4]: There's no doc page for `HEAD _index_template/{name}`, only for [`HEAD _template/{name}`](https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-template-exists-v1.html). I've considered linking to the `GET _index_template/{name}`, but instead to the general page on the topic, which is better than the old redirect page.

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.simulate_index_template.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "indices.simulate_index_template":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-templates.html",
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-simulate-template.html",
```
Comment (jrodewig) [inline rest-api-spec/src/main/resources/rest-api-spec/api/indices.simulate_index_template.json:4]:
```suggestion
      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-simulate-index.html",
```

**Review by jrodewig — COMMENTED** [review]
One minor correction. Otherwise LGTM.

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.simulate_index_template.json (hunk 2 above)
Comment (pquentin) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.simulate_index_template.json:4]: I don’t think that’s right, this API really simulates the template, not the index?

Comment (jrodewig) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.simulate_index_template.json:4]: Thanks for asking — I double-checked, and I think my suggestion is correct.

The API is named "Simulate index" (which is confusing IMO), but the path, HTTP method, and descriptions all line up. 

The purpose of the API is to test what index templates would be applied to a specific index name.

Comment (pquentin) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.simulate_index_template.json:4]: Thanks for taking the time to explain this, and sorry for the noise!

**Review by jrodewig — APPROVED** [review]

**Pull request merged**